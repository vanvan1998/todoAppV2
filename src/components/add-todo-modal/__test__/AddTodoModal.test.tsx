import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { AddTodoModal } from '..';

jest.mock('src/hooks', () => {
  const originalModule = jest.requireActual('src/hooks');
  return {
    ...originalModule,
    useMediaQuery: jest.fn().mockReturnValue({
      isMobile: true
    })
  };
});

describe('Test AddTodoModal component', () => {
  test('should match its snapshot', async () => {
    const component = render(<AddTodoModal addItem={jest.fn()} isButton />);
    expect(component).toMatchSnapshot();
  });
});
