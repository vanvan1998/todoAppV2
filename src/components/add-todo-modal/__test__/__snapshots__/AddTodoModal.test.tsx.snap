// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddTodoModal component should match its snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div>
      <div>
        <button
          buttontype="primary"
          class="sc-ixGGxD jZPfJ"
          data-testid="show-modal-button"
          style="height: 36px; min-height: 36px; padding: 0px 16px; display: flex; align-items: center; gap: 8px;"
          type="button"
        >
          Add new task
          <svg
            fill="currentColor"
            height="16"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="width: 100%; height: 100vh; padding-top: 16px; display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="sc-drVZOg bGlSty"
          >
            <div
              class="sc-eWPXlR buhMJJ"
            >
              <div
                class="sc-gtLWhw jWJCAr"
                styles="font-size: 20px; font-weight: 600;"
              >
                New Task Todo
              </div>
              <div
                class="sc-jIyAiq DlTWn"
              />
            </div>
            <div
              style="margin-bottom: 16px;"
            >
              <div
                class="sc-gtLWhw jPWJOW"
                style="font-size: 14px; font-weight: 500;"
              >
                Title Task
              </div>
              <input
                aria-label="Title Task"
                class="sc-epnzzT gwblLr"
                placeholder="Enter new task todo..."
                style="max-width: 438px; min-width: 294px; width: 100%;"
                type="text"
                value=""
              />
            </div>
            <div
              class="sc-gtLWhw tBlfZ"
              styles="font-weight: 500; margin-bottom: 8px;"
            >
              Category
            </div>
            <div
              class="sc-lnsxGb geneua"
            >
              <button
                buttontype="primary"
                class="sc-ixGGxD jZPfJ"
                data-testid="upcoming-button"
                style="height: 36px; min-height: 36px; padding: 0px 16px; display: flex; align-items: center; gap: 8px; width: 130px; background-color: rgb(100, 89, 227);"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="16"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm64-192c0-8.8 7.2-16 16-16h96c8.8 0 16 7.2 16 16v96c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16v-96zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
                  />
                </svg>
                Upcoming
              </button>
              <button
                buttontype="primary"
                class="sc-ixGGxD jZPfJ"
                data-testid="important-button"
                style="height: 36px; min-height: 36px; padding: 0px 16px; display: flex; align-items: center; gap: 8px; width: 130px; background-color: rgb(182, 180, 185);"
                type="button"
              >
                <svg
                  color="#d25a12"
                  fill="currentColor"
                  height="16"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.01 21.01c0 1.1.89 1.99 1.99 1.99s1.99-.89 1.99-1.99h-3.98zM12 6c2.76 0 5 2.24 5 5v7H7v-7c0-2.76 2.24-5 5-5zm0-4.5c-.83 0-1.5.67-1.5 1.5v1.17C7.36 4.85 5 7.65 5 11v6l-2 2v1h18v-1l-2-2v-6c0-3.35-2.36-6.15-5.5-6.83V3c0-.83-.67-1.5-1.5-1.5zM11 8h2v4h-2zm0 6h2v2h-2z"
                  />
                </svg>
                Important
              </button>
              <button
                buttontype="primary"
                class="sc-ixGGxD jZPfJ"
                data-testid="urgent-button"
                style="height: 36px; min-height: 36px; padding: 0px 16px; display: flex; align-items: center; gap: 8px; width: 130px; background-color: rgb(182, 180, 185);"
                type="button"
              >
                <svg
                  color="#db0f25"
                  fill="currentColor"
                  height="16"
                  stroke="currentColor"
                  stroke-width="0"
                  style="transform: rotate(270deg);"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="m3.5 18.99 11 .01c.67 0 1.27-.33 1.63-.84L20.5 12l-4.37-6.16c-.36-.51-.96-.84-1.63-.84l-11 .01L8.34 12 3.5 18.99z"
                  />
                </svg>
                Urgent
              </button>
            </div>
            <div
              class="sc-gtLWhw tBlfZ"
              styles="font-weight: 500; margin-bottom: 8px;"
            >
              Description
            </div>
            <textarea
              class="sc-eDHQDy dVNgLm"
              placeholder="Enter description..."
            />
            <div
              class="sc-jSSkKI gxzaZa"
            >
              <div
                class="sc-gtLWhw kuqrfP"
                styles="font-size: 14px; font-weight: 500;"
              >
                Notification
              </div>
              <div
                class="form-check form-switch"
              >
                <input
                  class="form-check-input"
                  id="flexSwitchCheckDefault"
                  role="switch"
                  style="width: 40px; height: 20px;"
                  type="checkbox"
                />
              </div>
            </div>
            <div
              class="sc-giBObj isODVq"
            >
              <button
                buttontype="secondary"
                class="sc-ixGGxD fjIgFe"
                data-testid="cancel-todo-button"
                style="flex: 1;"
                type="button"
              >
                Cancel
              </button>
              <button
                buttontype="primary"
                class="sc-ixGGxD jZPfJ"
                data-testid="add-todo-button"
                style="flex: 1;"
                type="button"
              >
                Create
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <button
        buttontype="primary"
        class="sc-ixGGxD jZPfJ"
        data-testid="show-modal-button"
        style="height: 36px; min-height: 36px; padding: 0px 16px; display: flex; align-items: center; gap: 8px;"
        type="button"
      >
        Add new task
        <svg
          fill="currentColor"
          height="16"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 277.333H277.333V416h-42.666V277.333H96v-42.666h138.667V96h42.666v138.667H416v42.666z"
          />
        </svg>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
